File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/UseBedPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record UseBedPacket(int entityId, int x, short y, int z) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeInt(x);
        os.writeByte(y);
        os.writeInt(z);
    }

    @Override
    public byte getId() {
        return 0x0A;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/UseBedPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class UseBedPacket(val entityId: Int, val x: Int, val y: Short, val z: Int) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId, x)
        os.writeByte(y.toInt())
        os.writeInt(z)
    }

    override val id: Byte
        get() = 0x0A
}
